AI/VCAI/FuzzyEngines.cpp

```diff
@@ -202,6 +202,8 @@ TacticalAdvantageEngine::TacticalAdvantageEngine()
 
 float TacticalAdvantageEngine::getTacticalAdvantage(const CArmedInstance * we, const CArmedInstance * enemy)
 {
+	boost::unique_lock<boost::mutex> lock(mx);
```

Comment:
It should be more clear why `getTacticalAdvantage` is synchronized while other fuzzy engines methods not.

Reply:
Made pathfinding having its own instance of FuzzyHelper. No need in mutex now